### PR TITLE
feat: add wake buyList query

### PR DESCRIPTION
**Pull request by IncognitaDev**
## What is this contribution about?

This PR adds kit query and return on PDP with kit products on isAccessoryOrSparePartOf

obs: although the kit itself its not a real product the fields of query are the same of product query, and this format are compatible with the way of vtex integration return a kit product.

### CODE REVIEW

File: wake/utils/transform.ts [hunk 1]
```diff
@@ -303,6 +303,17 @@ export const toProduct = (
     });
   }
 
+  //@ts-expect-error kit prop exists on buy list
+  if (variant.kit) {
```
Comment (matheusgr) [inline wake/utils/transform.ts:307]: Please don't ignore such type errors.

If you really need, set variant as a BuyList type (`variant: ProductFragment | SingleProductFragment | BuyList`), or work refactoring `toProduct` to deal with two distinct types.

**Review by matheusgr — CHANGES_REQUESTED** [review]

File: wake/utils/transform.ts (hunk 1 above)
Comment (IncognitaDev) [reply wake/utils/transform.ts]: I will refact this functions in other PR

File: wake/loaders/productDetailsPage.ts [hunk 2]
```diff
@@ -39,17 +42,59 @@ async function loader(
...
+          includeSameParent,
+        });
+
+      if (!buyListProductPage) return;
+
+      buyListProductPage.product.additionalProperty?.push({
+        "@type": "PropertyValue",
+        name: "quantity",
+        value: String(quantity),
+        valueReference: "BUY_LIST_PROPERTY",
```
[the remark this answers is not in the document]
Comment (matheusgr) [reply wake/loaders/productDetailsPage.ts]: @IncognitaDev still pending this fix.

name should not be quantity, but something that has semantic value to identify this property.

[the remark this answers is not in the document]
Comment (matheusgr) [reply wake/loaders/productDetailsPage.ts]: proposal: "SuggestedQuantity"

**Review by matheusgr — APPROVED** [review]

**Pull request merged**